src/armTemplates/resources/functionApp.test.ts

```diff
@@ -1,51 +1,126 @@
 import { FunctionAppResource } from "./functionApp";
-import { ServerlessAzureConfig } from "../../models/serverless";
+import { ServerlessAzureConfig, FunctionAppOS, SupportedRuntimeLanguage } from "../../models/serverless";
 
 describe("Function App Resource", () => {
+
   const resourceGroupName = "myResourceGroup";
   const prefix = "prefix";
   const region = "westus";
   const stage = "prod";
 
+  const defaultConfig: ServerlessAzureConfig = {
+    provider: {
+      name: "azure",
+      prefix,
+      region,
+      stage,
+      resourceGroup: resourceGroupName,
+      functionRuntime: {
+        language: SupportedRuntimeLanguage.NODE,
```

Comment:
Does this interfere with setting the os / runtime directly on the provider?  What are we recommending to users?  Why do we have 2 ways?

Reply:
This is an object that I'm inferring from the `runtime` in the provider, and then setting in `functionRuntime`. I didn't want to change `runtime` directly, but I can if that would be better. I just didn't want to parse and instantiate this object every time

Reply:
Will add as private configuration within config service

Reply:
I just realized that I can't do this ^. It needs to be inside the `ServerlessAzureConfig` because that's the only data that is passed to the ARM template resources like `functionApp`. Will add a comment on the property indicating that it's only for internal use